### PR TITLE
[Bugfix] TraceQL Metrics fix for edge case

### PR DESCRIPTION
**Pull request by ruslan-mikhailov**
**What this PR does**: Fixed an edge case bug: spans in a block with a single trace could not be queried if span's timestamp is rounded to seconds.

**Which issue(s) this PR fixes**:
Fixes #<issue number>

**Checklist**
- [ ] Tests updated
- [ ] Documentation added
- [ ] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: pkg/traceql/engine_metrics.go [hunk 1]
```diff
@@ -136,7 +136,9 @@ func IntervalOfMs(tsmills int64, start, end, step uint64) int {
 func TrimToBlockOverlap(start, end, step uint64, blockStart, blockEnd time.Time) (uint64, uint64, uint64) {
 	wasInstant := end-start == step
 
-	start2 := uint64(blockStart.UnixNano())
+	// We subtract 1 nanosecond from the block's start time
+	// to make sure we include the left border of the block.
+	start2 := uint64(blockStart.UnixNano()) - 1
```
Comment (ruslan-mikhailov) [inline pkg/traceql/engine_metrics.go:141]: Is it possible to have block's start time 0? In that case, it will have overflow here. Not sure, the case worths handling.

Comment (joe-elliott) [reply pkg/traceql/engine_metrics.go:141]: i think the answer is technically yes, but it would be quite difficult.

I would guess that a block start of 0 is more likely related to a bug then actually starting a block on second 0 of the unix epoch.

File: pkg/traceql/engine_metrics_test.go [hunk 2]
```diff
@@ -221,21 +221,21 @@ func TestTrimToBlockOverlap(t *testing.T) {
...
+			"2024-01-01 01:30:59", "2024-01-01 02:01:00", 5 * time.Minute,
 		},
 		{
 			// Instant query
 			// Original range is 1h
 			// Inner overlap is only 30m and step is updated to match
 			"2024-01-01 01:00:00", "2024-01-01 02:00:00", time.Hour,
 			"2024-01-01 01:30:00", "2024-01-01 02:30:00",
-			"2024-01-01 01:30:00", "2024-01-01 02:00:00", 30 * time.Minute,
+			"2024-01-01 01:29:59", "2024-01-01 02:00:00", 30*time.Minute + time.Nanosecond,
```
Comment (mdisibio) [inline pkg/traceql/engine_metrics_test.go:238]: Hmm this was unexpected, but it does seem correct.   01:30-02:00, is only 30 minutes when half open half closed borders.  Since we want the block start to be included, it really is + 1 ns.

Comment (ruslan-mikhailov) [reply pkg/traceql/engine_metrics_test.go:238]: Plus one ns I believe comes from changing step value for an instant query.
